lib/routes/feixiaohao/ranking.ts

```diff
@@ -0,0 +1,171 @@
+import { Route, ViewType } from '@/types';
+import ofetch from '@/utils/ofetch';
+
+export const route: Route = {
+    path: '/ranking/:type?',
+    categories: ['finance'],
+    view: ViewType.Articles,
+    example: '/feixiaohao/ranking/marketcap',
+    parameters: {
+        type: {
+            description: '排行榜类型',
+            default: 'marketcap',
+            options: [
+                { label: '市值排行', value: 'marketcap' },
+                { label: '涨幅排行', value: 'rise' },
+                { label: '跌幅排行', value: 'drop' },
+                { label: '成交量排行', value: 'volume' },
+                { label: '换手率排行', value: 'turnover' },
+                { label: '新币发行', value: 'newcoin' },
+            ],
+        },
+    },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['feixiaohao.com/', 'feixiaohao.com/hot/:type'],
+            target: '/ranking/:type',
+        },
+    ],
+    name: '币种排行榜',
+    url: 'feixiaohao.com',
+    maintainers: ['lijialin'],
+    handler,
+};
+
+async function handler(ctx) {
+    const type = ctx.req.param('type') ?? 'marketcap';
+    const baseUrl = 'https://www.feixiaohao.com';
+
+    // 排行榜类型映射
+    const typeMap = {
+        marketcap: { name: '市值排行', url: '/' },
+        rise: { name: '涨幅排行', url: '/hot/rise.html' },
+        drop: { name: '跌幅排行', url: '/hot/drop.html' },
+        volume: { name: '成交量排行', url: '/hot/currencies.html' },
+        turnover: { name: '换手率排行', url: '/hot/turnoverrate.html' },
+        newcoin: { name: '新币发行', url: '/hot/newcoin.html' },
+    };
+
+    const rankingInfo = typeMap[type] || typeMap.marketcap;
+    const url = `${baseUrl}${rankingInfo.url}`;
+
+    try {
+        await ofetch(url, {
```

Comment:
Please use the API endpoints at
`/api/coin/web-coinrank`
`/api/v2/coin/maxchange`
`/api/v2/ranking/coinvol`
`/api/v2/ranking/turnover`
`/api/home/web-newcoin`
instead